sklearn/cross_decomposition/pls_.py

```diff
@@ -31,7 +31,11 @@ def _nipals_twoblocks_inner_loop(X, Y, mode="A", max_iter=500, tol=1e-06,
     similar to the Power method for determining the eigenvectors and
     eigenvalues of a X'Y.
     """
-    y_score = Y[:, [0]]
+    for col in Y.T:
+        if np.any(np.abs(col) > np.finfo(np.double).eps):
```

Comment:
This is not checking if the column is constant. Perhaps use `np.ptp(Y, axis=0)` to find constant columns

Reply:
The purpose here is to check if the column has only zero values. The Y matrix that is passed to the function `_nipals_twoblocks_inner_loop` is a "centered" version of the Y matrix defined by the user.  The "centering" is performed in the function `_center_scale_xy` and one of the operations it performs is the following:
```
y_mean = Y.mean(axis=0)
Y -= y_mean
```
If the original Y contains a constant column, then the resulting Y will contain a column of zeros, and this is what it is causing the problem in the `_nipals_twoblocks_inner_loop`